t/005-apicast-oauth.t

```diff
@@ -361,3 +362,38 @@ GET /?access_token=foobar
 --- error_code: 200
 --- response_body
 yay, upstream
+
+=== TEST 12: calling /authorize with state returns same value back on redirect_uri
```

Comment:
I'd say test 2 and 3 are quite similar to what you want, no? Just not checking the state parameter.